src/tabmat/categorical_matrix.py

```diff
@@ -540,6 +599,16 @@ def multiply(self, other) -> sps.csr_matrix:
                 f"Shapes do not match. Expected length of {self.shape[0]}. Got {len(other)}."
             )
 
+        if self.drop_first:
+            return sps.csr_matrix(
+                (
+                    np.squeeze(other),
+                    self.indices,
+                    np.arange(self.shape[0] + 1, dtype=int),
+                ),
+                shape=(self.shape[0], self.shape[1] + 1),
+            )[:, 1:]
```

Comment:
```suggestion
                shape=(self.shape[0], self.shape[1] + self.drop_first),
            )[:, self.drop_first :]
```
and collapse the two blocks?

Reply:
same thing here. If we can avoid slicing it would be ideal, so I keep them separate.